src/core/thread/osthread.d

```diff
@@ -3189,6 +3179,23 @@ private void onThreadError(string msg) nothrow @nogc
     throw error;
 }
 
+version (Posix)
+private size_t adjustStackSize(size_t sz) nothrow @nogc
+{
+    if (sz == 0)
+        return 0;
+
+    // TLS uses the top of the stack, so add its size to the requested size
+    sz += externDFunc!("rt.sections_elf_shared.sizeOfTLS",
+                       size_t function() @nogc nothrow)();
```

Comment:
Isn't it better to add `version(CRuntime_Glibc)` here?

Reply:
Probably, as FreeBsd doesn't seem to have this issue. I've added the version check.